static/app/views/dashboardsV2/controls.tsx

```diff
@@ -118,17 +121,25 @@ class Controls extends React.Component<Props> {
                 {t('Edit Dashboard')}
               </Button>
               {organization.features.includes('widget-library') ? (
-                <Button
-                  data-test-id="add-widget-library"
-                  priority="primary"
-                  icon={<IconAdd isCircled />}
-                  onClick={e => {
-                    e.preventDefault();
-                    onAddWidget();
-                  }}
+                <Tooltip
+                  title={tct('Max widgets ([maxWidgets]) per dashboard reached.', {
+                    maxWidgets: MAX_WIDGETS,
+                  })}
+                  disabled={!!!(widgetCount >= MAX_WIDGETS)}
                 >
-                  {t('Add Widget')}
-                </Button>
+                  <Button
+                    data-test-id="add-widget-library"
+                    priority="primary"
+                    disabled={widgetCount >= MAX_WIDGETS}
+                    icon={<IconAdd isCircled />}
+                    onClick={e => {
```

Comment:
Do we need to `e.preventDefault` if this isn't part of a form? Preventing a form submit is usually why I see it, is there another reason we have it here? Otherwise we could just make this `onClick={onAddWidget}` :D

Reply:
dang I thought it was required for all the onClicks, TIL

Reply:
yeah...I added this because a lot of the buttons around dashboards used the same pattern but I think you're right. It doesn't seem necessary. Also thanks for making me look into preventDefault because I realized I only had a partial understanding of what it did lol.

Reply:
0ec681ce149283010f2f467ce8b21995ca4cad9a